parser_library/src/context/instruction.cpp

```diff
@@ -2096,985 +2146,1046 @@ constexpr auto mi_VUPLL = find_mi("VUPLL");
-    { "WFLNDB", mi_VFPSO, { { 2, 3 }, { 3, 8 }, { 4, 1 } } },
-    { "WFLNSB", mi_VFPSO, { { 2, 2 }, { 3, 8 }, { 4, 1 } } },
-    { "WFLNXB", mi_VFPSO, { { 2, 4 }, { 3, 8 }, { 4, 1 } } },
-    { "WFLPDB", mi_VFPSO, { { 2, 3 }, { 3, 8 }, { 4, 2 } } },
-    { "WFLPSB", mi_VFPSO, { { 2, 2 }, { 3, 8 }, { 4, 2 } } },
-    { "WFLPXB", mi_VFPSO, { { 2, 4 }, { 3, 8 }, { 4, 2 } } },
-    { "WFLRD", mi_VFLR, { { 2, 3 } } }, // operand with index 3 ORed with 8
-    { "WFLRX", mi_VFLR, { { 2, 4 } } }, // operand with index 3 ORed with 8
-    { "WFMADB", mi_VFMA, { { 4, 8 }, { 5, 3 } } },
-    { "WFMASB", mi_VFMA, { { 4, 8 }, { 5, 2 } } },
-    { "WFMAXB", mi_VFMA, { { 4, 8 }, { 5, 4 } } },
-    { "WFMAXDB", mi_VFMAX, { { 3, 3 }, { 4, 8 } } },
-    { "WFMAXSB", mi_VFMAX, { { 3, 2 }, { 4, 8 } } },
-    { "WFMAXXB", mi_VFMAX, { { 3, 4 }, { 4, 8 } } },
-    { "WFMDB", mi_VFM, { { 3, 3 }, { 4, 8 } } },
-    { "WFMINDB", mi_VFMIN, { { 3, 3 }, { 4, 8 } } },
-    { "WFMINSB", mi_VFMIN, { { 3, 2 }, { 4, 8 } } },
-    { "WFMINXB", mi_VFMIN, { { 3, 4 }, { 4, 8 } } },
-    { "WFMSB", mi_VFM, { { 3, 2 }, { 4, 8 } } },
-    { "WFMSDB", mi_VFMS, { { 4, 8 }, { 5, 3 } } },
-    { "WFMSSB", mi_VFMS, { { 4, 8 }, { 5, 2 } } },
-    { "WFMSXB", mi_VFMS, { { 4, 8 }, { 5, 4 } } },
-    { "WFMXB", mi_VFM, { { 3, 4 }, { 4, 8 } } },
-    { "WFNMADB", mi_VFNMA, { { 4, 8 }, { 5, 3 } } },
-    { "WFNMASB", mi_VFNMA, { { 4, 8 }, { 5, 2 } } },
-    { "WFNMAXB", mi_VFNMA, { { 4, 8 }, { 5, 4 } } },
-    { "WFNMSDB", mi_VFNMS, { { 4, 8 }, { 5, 3 } } },
-    { "WFNMSSB", mi_VFNMS, { { 4, 8 }, { 5, 2 } } },
-    { "WFNMSXB", mi_VFNMS, { { 4, 8 }, { 5, 4 } } },
-    { "WFPSODB", mi_VFPSO, { { 2, 3 }, { 3, 8 } } },
-    { "WFPSOSB", mi_VFPSO, { { 2, 2 }, { 3, 8 } } },
-    { "WFPSOXB", mi_VFPSO, { { 2, 4 }, { 3, 8 } } },
-    { "WFSDB", mi_VFS, { { 2, 3 }, { 3, 8 } } },
-    { "WFSQDB", mi_VFSQ, { { 2, 3 }, { 3, 8 } } },
-    { "WFSQSB", mi_VFSQ, { { 2, 2 }, { 3, 8 } } },
-    { "WFSQXB", mi_VFSQ, { { 2, 4 }, { 3, 8 } } },
-    { "WFSSB", mi_VFS, { { 2, 2 }, { 3, 8 } } },
-    { "WFSXB", mi_VFS, { { 2, 4 }, { 3, 8 } } },
-    { "WFTCIDB", mi_VFTCI, { { 3, 3 }, { 4, 8 } } },
-    { "WFTCISB", mi_VFTCI, { { 3, 2 }, { 4, 8 } } },
-    { "WFTCIXB", mi_VFTCI, { { 3, 4 }, { 4, 8 } } },
-    { "WLDEB", mi_VFLL, { { 2, 2 }, { 3, 8 } } },
-    { "WLEDB", mi_VFLR, { { 2, 3 } } }, // operand with index 3 ORed with 8
-    { "XHHR", mi_RXSBG, { { 2, 0 }, { 3, 31 } } },
-    { "XHHR", mi_RXSBG, { { 2, 0 }, { 3, 31 } } },
-    { "XHLR", mi_RXSBG, { { 2, 0 }, { 3, 31 }, { 4, 32 } } },
-    { "XLHR", mi_RXSBG, { { 2, 32 }, { 3, 63 }, { 4, 32 } } },
-    { "XLHR", mi_RXSBG, { { 2, 32 }, { 3, 63 }, { 4, 32 } } },
+// clang-format off
+constexpr std::pair<mnemonic_code, supported_system> mnemonic_codes[] = {
+    { { "B", mi_BC, { { 0, 15 } } }, { supported_system::UNI | supported_system::ESA | supported_system::XA | supported_system::_370 | supported_system::DOS | supported_system::SINCE_ZOP } }, 
```

Comment:
the more I think about it, the more I am convinced that the mnemonic's availability is determined by the availability of the underlaying instruction. So this should not be needed at all.

Reply:
I've double checked this and unfortunately it seems that some mnemonics were added later than their machine instruction counterparts...

Reply:
interesting...